src/map.c

```diff
@@ -479,17 +489,16 @@ static void set_char(char *data, int linesz, int nlines, int x, int z, char c, i
 void map_save(const struct Map *map)
 {
 	SDL_assert(map->custom);
-	int linesz = strlen("|--")*map->xsize + strlen("|\n");
+	int linesz = strlen("|--")*map->xsize + strlen("|") + sizeof("");
```

Comment:
A bit less explicit, but also doesn't look like a typo

```suggestion
	int linesz = strlen("|--")*map->xsize + strlen("|") + 1;
```